lib/charms/mysql/v0/mysql.py

```diff
@@ -2524,6 +2548,20 @@ def get_pid_of_port_3306(self) -> Optional[str]:
         except MySQLExecError:
             return None
 
+    def flush_mysql_logs(self, logs_type: MySQLTextLogs) -> None:
+        """Flushes the specified logs_type logs."""
+        flush_logs_commands = (
+            f"shell.connect('{self.server_config_user}:{self.server_config_password}@{self.instance_address}')",
+            'session.run_sql("SET sql_log_bin = 0")',
+            f'session.run_sql("FLUSH {logs_type.value}")',
+            'session.run_sql("SET sql_log_bin = 1")',
```

Comment:
nit: it is not necessary, sql_bin_log is a session parameter.

Reply:
will address nit in a separate PR